java/integTest/build.gradle

```diff
@@ -155,7 +155,56 @@ tasks.withType(Test) {
     afterTest { desc, result ->
         logger.quiet "${desc.className}.${desc.name}: ${result.resultType} ${(result.getEndTime() - result.getStartTime())/1000}s"
     }
+
+    finalizedBy jacocoTestReport, jacocoTestCoverageVerification
+}
+
+// JaCoCo section (code coverage by integration tests)
+jacocoTestReport {
+    reports {
+        xml.configure { enabled false }
+        csv.configure { enabled false }
+        html.configure { enabled true } 
+    }
+
+    sourceDirectories.setFrom(files("${project.rootDir}/client/src/main/java"))
+    classDirectories.setFrom(files("$buildDir/classes"))
+    executionData.setFrom(
+        files("$buildDir/jacoco/test.exec"),
+        files("$buildDir/jacoco/pubsubTest.exec")
+    )
+    
+    afterEvaluate {
+        classDirectories.setFrom(files(classDirectories.files.collect {
+            fileTree(dir: it)
+        }))
+    }
+    
+}
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            element = 'CLASS'
+            excludes = [
+                'glide.models.protobuf.*'
```

Comment:
Please, add 'glide.com.google.*' (shadowed protobuf) and 'glide.io.netty.*' (shadowed netty) and a comment